core/src/avm2/globals/flash/display/movieclip.rs

```diff
@@ -24,17 +22,6 @@ pub fn instance_init<'gc>(
 ) -> Result<Value<'gc>, Error<'gc>> {
     if let Some(this) = this {
         activation.super_init(this, &[])?;
-
-        if this.as_display_object().is_none() {
-            let class_object = this
-                .instance_of()
-                .ok_or("Attempted to construct MovieClip on a bare object")?;
-            let movie = Arc::new(SwfMovie::empty(activation.context.swf.version()));
-            let new_do =
-                MovieClip::new_with_avm2(movie, this, class_object, activation.context.gc_context);
```

Comment:
Could it be that `MovieClip::new_with_avm2` is no longer used? If so, we can remove it.

Reply:
It's still used in `sprite`

Reply:
Oh well...